jobs/webcompat-kb/webcompat_kb/metric_changes.py

```diff
@@ -0,0 +1,565 @@
+            elif "webcompat:site-report" in change.removed:
+                reasons.add("site-report-removed")
+        elif change.field_name == "status":
+            if change.added in FIXED_STATES and change.removed not in FIXED_STATES:
+                reasons.add("resolved")
+            elif change.removed in FIXED_STATES and change.added not in FIXED_STATES:
+                reasons.add("reopened")
+
+    rv = list(reasons)
+    rv.sort()
+    return rv
+
+
+def compute_score_changes(
+    changes_by_bug: Mapping[int, list[BugChange]],
+    bug_data: Mapping[int, BugData],
+    historic_states: Mapping[int, list[BugState]],
+    historic_scores: Mapping[int, list[float]],
+    last_change_time: datetime,
+) -> Mapping[int, list[ScoreChange]]:
+    rv: dict[int, list[ScoreChange]] = {}
+
+    for bug_id, states in historic_states.items():
+        rv[bug_id] = []
+
+        scores = historic_scores[bug_id]
+        changes = changes_by_bug.get(bug_id)
+        bug = bug_data[bug_id]
+        newly_created = bug.creation_time > last_change_time
+
+        assert len(states) == len(scores)
+
+        prev_score = None if not newly_created else 0.0
+
+        for state, score in zip(reversed(states), reversed(scores)):
+            if prev_score is None:
+                score_change = ScoreChange(
+                    who=bug.creator,
+                    change_time=bug.creation_time,
+                    score_delta=score,
+                    reasons=["created"],
+                )
+            elif state.change_idx is not None:
+                assert changes is not None
+                score_delta = score - prev_score
+                change = changes[state.change_idx]
+                reasons = get_change_reasons(change.changes)
+                if not reasons and score_delta > 0:
+                    logging.warning(
+                        f"No change reason for {bug_id} with change {change.changes}"
+                    )
+
+                score_change = ScoreChange(
+                    who=change.who,
+                    change_time=change.change_time,
+                    score_delta=score_delta,
+                    reasons=reasons,
+                )
+            else:
+                assert state == states[-1]
+                score_change = None
+
+            prev_score = score
+            if score_change is not None and score_change.score_delta != 0:
+                rv[bug_id].append(score_change)
+
+    changed_bugs = {key: len(value) for key, value in rv.items() if len(value)}
+    logging.info(
+        f"Got {sum(changed_bugs.values())} score changes in {len(changed_bugs)} bugs"
+    )
+
+    #         if len(rv[bug_id]) == 0 and any(is_webcompat_bug(state.product, state.component, state.keywords) for state in states):
+    #             logging.info(f"""No score changes for WebCompat bug {bug_id}:
+    #   STATES:  {states}
+    #   CHANGES: {changes}
+    #   SCORES:  {scores}
+    # """)
```

Comment:
Is there a reason why this is left-in, but commented out? Seems like useful info to throw into `logging.debug`.